exercises/robot-simulator/tests/robot-simulator.rs

```diff
@@ -0,0 +1,147 @@
+extern crate robot_simulator;
+
+use robot_simulator::*;
+
+#[test]
+fn robots_are_created_with_position_and_direction() {
+    let robot = Robot::new(0, 0, &Direction::North);
+    assert_eq!((0, 0), robot.position());
+    assert_eq!(&Direction::North, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn positions_can_be_negative() {
+    let robot = Robot::new(-1, -1, &Direction::South);
+    assert_eq!((-1, -1), robot.position());
+    assert_eq!(&Direction::South, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_does_not_change_position() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_right();
+    assert_eq!((0, 0), robot.position());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_north_points_the_robot_east() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_right();
+    assert_eq!(&Direction::East, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_east_points_the_robot_south() {
+    let robot = Robot::new(0, 0, &Direction::East).turn_right();
+    assert_eq!(&Direction::South, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_south_points_the_robot_west() {
+    let robot = Robot::new(0, 0, &Direction::South).turn_right();
+    assert_eq!(&Direction::West, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_west_points_the_robot_north() {
+    let robot = Robot::new(0, 0, &Direction::West).turn_right();
+    assert_eq!(&Direction::North, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_left_does_not_change_position() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_left();
+    assert_eq!((0, 0), robot.position());
+}
+
+#[test]
+#[ignore]
+fn turning_left_from_north_points_the_robot_west() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_left();
```

Comment:
```rust
let robot = Robot::new(..);
let robot = robot.turn_left();
assert ...
```
If these tests were written in the above form then users might be able to implement the methods with either of these signatures: 
- `fn builder_style(self) -> Self`
- `fn with_copies(&self) -> Self` 
- `fn mutably(&mut self) -> &mut Self`